src/TorchSharp/Functional.cs

```diff
@@ -0,0 +1,114 @@
+                {
+                    List<Tensor> result = new List<Tensor>();
+                    foreach (var input in inputs)
+                    {
+                        result.Add(input.detach().requires_grad_(need_graph));
+                    }
+                    return result.ToArray();
+                }
+
+                /// <summary>
+                /// Wrapper function around torch.autograd.grad.
+                /// </summary>
+                /// <param name="output">The output tensor to compute gradients for.</param>
+                /// <param name="retain_graph">Whether the gradient computation graph should be kept.</param>
+                /// <param name="inputs">The input tensors at which to evaluate the gradients.</param>
+                /// <returns>A collection of gradient tensors for the output with respect to each input.</returns>
+                private static IEnumerable<Tensor> _autograd_grad(Tensor output, bool retain_graph=false, params Tensor[] inputs)
+                {
+                    var outputs = new List<Tensor>
+                    {
+                        output
+                    };
+                    var inputList = new List<torch.Tensor>(inputs);
+                    return torch.autograd.grad(outputs, inputList, retain_graph: retain_graph);
+                }
+
+                /// <summary>
+                /// Computes the jacobian of a given function.
+                /// 
+                /// Example:
+                /// torch.Tensor x1 = torch.tensor(new double[,] { { 1.0, 3.0 }, { 2.0, 4.0 } }, requires_grad: true);
+                /// torch.Tensor intercepts = torch.tensor(new double[] { 1.0, 5.0 }, requires_grad: true);
+                /// 
+                /// torch.Tensor[] jacFunc(torch.Tensor[] inputs)
+                /// {
+                ///     return new torch.Tensor[] { torch.einsum("ij,j->i", inputs[0], inputs[1]) };
+                /// }
+                ///
+                /// var jacobian = torch.autograd.functional.jacobian(jacFunc, x1, intercepts);
+                ///
+                /// jacobian[0] should be:
+                /// [[[1, 5],
+                ///   [0, 0]],
+                ///  [[0, 0],
+                ///   [1, 5]]]
+                /// And jacobian[1]:
+                /// [[1, 3],
+                ///  [2, 4]]
+                ///  
+                /// </summary>
+                /// <param name="function">The function to compute the jacobian for. It may have multiple inputs and multiple outputs.</param>
+                /// <param name="inputs">The values for the inputs at which to compute the jacobian.</param>
+                /// <returns>A list of tensors, one for each output value of each output. For example, a single output with two elements yields two jacobian tensors.</returns>
+                public static IEnumerable<Tensor> jacobian(Func<Tensor[], Tensor[]> function, params Tensor[] inputs)
```

Comment:
This is very general, but consider whether there are other overloads that would be useful here.